specify/specify7
# Populate rankid with next enforced rank when adding tree node

Fixes #3597

### Testing Instructions
- Click on an existing tree node on any tree
- Use the `Add Child` tree action (displayed as a :heavy_plus_sign:)
- The rank (definitionitem) picklist should automatically populate with the next enforced rank in the tree

If is it unknown which ranks are enforced, perform a query on `{TREE}treedefitem` where {TREE} is the name of the tree (i.e. taxontreedefitem) and add the `isenforced` field to the query. 

Untill #4266 is merged, if there is more than one discipline, it would be helpful to narrow the results to tree items only in the desired discipline. 
This can be accomplished through `treedef -> discipline` when {TREE}treedefitem is the base table.
